pkg/template/values.go

```diff
@@ -18,8 +18,9 @@ import (
 type Values struct {
 	ValuesFrom []v1alpha1.AppTemplateValuesSource
 
-	appContext AppContext
-	coreClient kubernetes.Interface
+	appContext                  AppContext
+	coreClient                  kubernetes.Interface
+	additionalDownwardAPIValues AdditionalDownwardAPIValues
 }
 
 func (t Values) AsPaths(dirPath string) ([]string, func(), error) {
```

Comment:
Just a comment about this function and maybe not be actionable in this PR.
This signature feels strange, the main reason for that is the return of that `func()` that you need to read the code to understand that it is the removal of the created folders.
It is the responsibility of the creator of `dirPath` to delete that folder, while this function is only responsible for creating the folders it needs inside the `dirPath`.

Reply:
Not going to action on this one in this PR